Sql.py

```diff
@@ -65,30 +66,57 @@ def clear_table(self, table_name) -> None:
     def close(self) -> None:
         self._db.close()
 
-    @staticmethod
-    def _type_conversion_to_sql(type_) -> str:
+    def _type_conversion_to_sql(self, type_) -> str:
         if type_ == SchemaType.STR:
             return 'text'
         if type_ == SchemaType.INT:
             return 'integer'
         raise AssertionError('Unknown type "{}"'.format(type_))
 
-    @staticmethod
-    def _value_conversion_to_sql(value, type_) -> str:
+    def _value_conversion_to_sql(self, value, type_) -> str:
         if type_ == SchemaType.STR:
             return '"{}"'.format(value)
         if type_ == SchemaType.INT:
             return str(value)
         raise AssertionError('Unknown type "{}"'.format(type_))
 
-    @staticmethod
-    def _value_conversion_from_sql(value, type_) -> Union[str, int]:
+    def _type_conversion_from_sql(self, type_) -> Union[str, int]:
+        if type_ == 'text':
+            return SchemaType.STR
+        if type_ in ('integer', 'int'):
+            return SchemaType.INT
+        raise AssertionError('Unknown type "{}"'.format(type_))
+
+    def _value_conversion_from_sql(self, value, type_) -> Union[str, int]:
         if type_ == SchemaType.STR:
             return value
         if type_ == SchemaType.INT:
             return value
         raise AssertionError('Unknown type "{}"'.format(type_))
 
+    def _fill_in_schemas(self) -> None:
+        table_names = self._get_existed_table_names()
+        for table_name in table_names:
+            self._schemas[table_name] = self._get_schema_from_existed_table(table_name)
+
+    def _get_existed_table_names(self) -> list:
+        table_names = []
+        execution = 'select name from sqlite_master where type="table"'
+        for table_name in self._cursor.execute(execution):
+            table_names.append(table_name[0])
+        return table_names
+
+    def _get_schema_from_existed_table(self, table_name) -> dict:
+        schema = {}
+        execution = 'select sql from sqlite_master \
+            where type="table" and name="{}"'.format(table_name)
+        raw_schema = next(self._cursor.execute(execution))[0]
+        labels = raw_schema[(raw_schema.find('(') + 1):raw_schema.find(')')].split(',')
```

Comment:
可以先不用改，但有機會你可以玩玩看regular expression


```suggestion
        labels = re.search('\((.*?)\)', raw_schema).group(1).split(',')
```

Reply:
OK我看懂了這個re了，不過我測了一下好像還是原本的寫法比較快
![image](https://user-images.githubusercontent.com/37433310/131314720-9b299bc6-9ae2-4a62-9052-e4ca59098b9d.png)